### PR TITLE
Issue/24

### LINKED ISSUE #24: Radar
Radar system that can show off the whole map in the corner. Outline of player/enemies can be colored dots on the screen.

EDIT:
Minimap for corner of screen that displays information to the player about where the enemies are in relation to the play area and the farmer.

BASE REQUIREMENTS:
- Minimap size and aspect ratio will mirror that of the gameplay (fenced in) area (whole play area always represented, not just a perimeter around the player)
- Minimap will use a separate orthographic camera to overlay the gameplay camera
- Player represented by a yellow dot (slightly bigger than enemy dots)
- Stabber bunnies represented by red dots
- Nabber bunnies represented by orange dots
- Minimap will be aligned with the screen's rectangle, not the camera angle (which is going to be roughly 30-45 degrees angled)
- The dots will only represent positions of the players, not any other information

SIMPLIFIED DESIGN (based on 1920x1080 resolution)
- Dark grey background at approximately 15-25% opacity
- Background stroked on the edges with light grey of 2-4 pixels, at approximately 25-50% opacity
- Farmer Dot: Approximately 16 pixels in size
- Enemy Dot: Approximately 8 pixels in size

- Dots possibly full opacity, but if they feel like they are in the way, then as low as 50% opacity

NICE TO HAVE:
- Possibly have a VERY faded version of the play area, top down orthographic camera as a background instead of standard grey
- OR Possibly have standard grey, but with artistic landmarks in the corners of the minimap (meteorite, well, shed, gate) to help represent the play area

NOTES:
- Colors of the dots subject to change
- See attached image for basic idea
- Due to virtual thumbsticks for mobile, I'm guessing it will be best to place the minimap in one of the upper corners of the screen.  That said, if possible, maybe we can keep it in a lower corner by default, but then move it to either upper corner or even bottom-middle when using virtual sticks.  Certainly this may need playtesting if we do it.

MinimapExample

### PR DESCRIPTION
**Pull request by SweetAlmighty**
Closes #24

### CODE REVIEW

**Comment by ericbroberic** [comment]
@SweetAlmighty The rotation of the Farmer is off when moving off of the world origin. It is most noticeable if you move to the top left of the garden, the rotation of the Farmer is not following the mouse.

**Comment by ericbroberic** [comment]
@SweetAlmighty updateBlip is also throwing an error if you hit the end of the round.

**Comment by SweetAlmighty** [comment]
@ericbroberic I'm not seeing an error locally.

As for the rotation issue, that has always been there, at least on builds/branches that I've tested locally. Just tried a build on a different computer too. If you can point to how it is an issue with this branch, I'll fix it.

Edit: Never mind, I misread the issue initially with the error.

**Comment by ericbroberic** [comment]
@SweetAlmighty 

Latest build from develop: https://storage.googleapis.com/farmer-hosting/branches/develop/index.html

Latest build from this PR: https://storage.googleapis.com/farmer-hosting/prs/104/index.html

In the develop branch the Farmer always has the same rotation as the mouse (there is a bug where if the mouse isn't moving the Farmer rotation doesn't update, but that's different). In the PR branch, the Farmer does not always aim in the direction of the mouse, and has a weird behavior. I notice it the most when up on the top left of the garden, by the shed.

Let me know if you still need more info, I might be able to get you a video.

**Comment by SweetAlmighty** [comment]
@ericbroberic Okay, I see what you are talking about now. I thought you were talking about the bug you mentioned in your last comment.

I'll look into it.

File: src/ui/radar.ts [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+    }
+
+    /**
+     * Update function for radar blips.
+     * @param root The root TransformNode to retrieve the position from.
+     */
+    public static updateBlip(root: TransformNode): void {
+        const node = this.getInstance().#_blips.get(root);
+        if (node != null) {
+            node.position = new Vector3(root.position.x, node.position.y, root.position.z)
```
Comment (ericbroberic) [inline src/ui/radar.ts:142]: Pro tip: Vector3 has a clone function to create a new instance.

**Review by ericbroberic — APPROVED** [review]

**Pull request merged**